src/particles/elements/RFCavity.H

```diff
@@ -42,23 +89,49 @@ namespace impactx
          * @param escale Scaling factor for on-axis RF electric field in 1/m
          * @param freq RF frequency in Hz
          * @param phase RF driven phase in deg
+         * @param cos_coef TODO
+         * @param sin_coef TODO
          * @param mapsteps number of integration steps per slice used for
          *        map and reference particle push in applied fields
          * @param nslice number of slices used for the application of space charge
          */
+        AMREX_GPU_HOST
         RFCavity(
             amrex::ParticleReal const ds,
             amrex::ParticleReal const escale,
             amrex::ParticleReal const freq,
             amrex::ParticleReal const phase,
-            int const mapsteps,
-            int const nslice
+            std::vector<amrex::ParticleReal> cos_coef,
+            std::vector<amrex::ParticleReal> sin_coef,
+            int mapsteps = 1,
+            int nslice = 1
         )
           : Thick(ds, nslice),
             m_escale(escale), m_freq(freq), m_phase(phase), m_mapsteps(mapsteps)
         {
+            m_cos_size = cos_coef.size();
+            m_sin_size = sin_coef.size();
+            if (m_cos_size != m_sin_size)
+                throw std::runtime_error("RFCavity: cos and sin coefficients must have same length!");
+
+            m_cos_coef.resize(m_cos_size);
+            m_sin_coef.resize(m_sin_size);
+            amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice,
+                                  cos_coef.begin(), cos_coef.end(),
+                                  m_cos_coef.begin());
+            amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice,
+                                  sin_coef.begin(), sin_coef.end(),
+                                  m_sin_coef.begin());
+            amrex::Gpu::synchronize();
+
+            // low-level objects we can use on device
+            m_cos_data = m_cos_coef.data();
+            m_sin_data = m_sin_coef.data();
         }
 
+        AMREX_GPU_HOST
```

Comment:
This is a bit of a hack to avoid trying to call the destructor on GPU at the end of the kernel, with respect to the `m_cost_coeff`/`m_sin...` data members.

The reason is that we currently copy the whole element on the GPU in `beamoptic.H` - `PushSingleParticle`.

Nonetheless, this runs well on GPU (tested on Nvidia) and I'll clean this up later with a better design.